internal/controller/apiextensions/definition/watch.go

```diff
@@ -0,0 +1,32 @@
+package definition
+
+import (
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	v1 "github.com/crossplane/crossplane/apis/apiextensions/v1"
+	"github.com/crossplane/crossplane/internal/xcrd"
+)
+
+// OffersCompositeResource accepts any CompositeResourceDefinition or a
+// CustomResourceDefinition that represents a composite.
+func OffersCompositeResource() resource.PredicateFn {
```

Comment:
Interesting, I didn't actually know you could specify a predicate that applied to both the `For` and `Owns` objects. I think we've always used a separate `WithPredicates` argument to `For`/`Own`.

It may be worth doing that here? I don't feel strongly - the only advantage would be consistency and in the case of CRDs avoiding any cost of the XRD type assertion that will never be true.

Reply:
That makes sense to me; I have updated the code to match what you suggested.